docs/decisions/0007-sys-path-modification-removal.rst

```diff
@@ -32,11 +34,190 @@ This deprecation will take place in the following steps:
 
 4. Fix all instances where the ``sys.path``-based modules were ``patch``-ed in unit tests, as those patches no longer work.
 
-5. Once all instances of the ``sys.path``-based imports have been migrated in the ``edx-platform`` codebase, officially deprecate that import pattern for the next open-source release (as per the standard deprecation process).
+5. Once all instances of the ``sys.path``-based imports have been migrated in the ``edx-platform`` codebase, officially deprecate that import pattern for the next open-source release.
 
 6. Monitor logs to address continued use of the ``sys.path`` based import patterns.
 
 Goals
 =====
 
 Eliminate modifications to ``sys.path`` to enable easier tool use in edx-platform.
+
+
+Timeline
+========
+
+In the Koa release, usages of the old import style raised instances of
+``DeprecatedEdxPlatformImportWarning``, but still worked.
+
+In the Lilac release, usages old import paths will raise instances of
+``DeprecatedEdxPlatformImportError``, breaking any code that was not updated.
+This error class is intentionally *not* a subclass of ``ImportError``;
+to understand why, consider the following common pattern used in packages outside
+of the edx-platform repo itself::
+  try:
+      from student.models import CourseEnrollment
+  except ImportError:
+      CourseEnrollment = None
+This pattern is (unfortunately) widely used to so packages can import
```

Comment:
> (unfortunately)

I'm injecting my opinion here only to make it clear that I'm not encouraging others to follow this import pattern when making new edx-platform plugins :)